### PR TITLE
Collection page and minor fixes

### PR DESCRIPTION
**Pull request by UladBohdan**
Collection page: https://dev.wir.by/be/collection/starbellit/

Also:
* podcasts buttons in footer
* style fixes
* exception for `starbellit` collection
* fix for image alignment in articles

### CODE REVIEW

**Comment by UladBohdan** [comment]
[add podcasts links to footer](https://trello.com/c/X6g9ZJc3/507-add-podcasts-links-to-footer)

**Comment by UladBohdan** [comment]
[some style fixes](https://trello.com/c/2CmUaHVc/508-some-style-fixes)

**Comment by UladBohdan** [comment]
[remove cover from cards of collection старажытнабеларуская літаратура (make collection an exception)](https://trello.com/c/hxxq1TGb/510-remove-cover-from-cards-of-collection-%D1%81%D1%82%D0%B0%D1%80%D0%B0%D0%B6%D1%8B%D1%82%D0%BD%D0%B0%D0%B1%D0%B5%D0%BB%D0%B0%D1%80%D1%83%D1%81%D0%BA%D0%B0%D1%8F-%D0%BB%D1%96%D1%82%D0%B0%D1%80%D0%B0%D1%82%D1%83%D1%80%D0%B0-make-collection-an-exception)

**Comment by UladBohdan** [comment]
[implement collection page](https://trello.com/c/7Wc6wncT/506-implement-collection-page)

File: components/articles/cards/ArticleCard.js [hunk 1]
```diff
@@ -68,6 +68,10 @@ const getCoverLink = (images, cardSize) => {
 
 const COLLECTION_LOGO_WIDTH = 70;
 
+const COLLECTIONS_HIDE_COVER = {
+  starbellit: true,
+};
```
Comment (Drapegnik) [inline components/articles/cards/ArticleCard.js:73]: is there any cleaner solution?

File: components/articles/cards/ArticleCard.js [hunk 2]
```diff
@@ -158,7 +162,7 @@ const ArticleCard = props => {
         {!collection && (
           <div className={b('description-container')}>
             <div className={b('description')}>{subtitle}</div>
-            <div className={linkCn({ dark: theme === 'light' })}>
+            <div className={linkCn({ dark: theme !== 'light' })}>
```
Comment (Drapegnik) [inline components/articles/cards/ArticleCard.js]:
```js
const dark = theme !== 'light';

linkCn({ dark })
```

File: components/common/layout/Footer.js [hunk 3]
```diff
@@ -29,19 +26,19 @@ const Footer = () => (
...
+            // `themes` should be first
             {TOPICS.map(topic => (
               <div key={topic} className={f('item')}>
                 <Link route={ROUTES_NAMES.topic} params={{ topic }}>
                   <Text id={`topic.by_${topic}`} />
                 </Link>
               </div>
             ))}
-          </div>
+          </div> */}
```
Comment (Drapegnik) [inline components/common/layout/Footer.js:41]: ?

File: components/common/layout/Footer.js [hunk 4]
```diff
@@ -54,6 +51,18 @@ const Footer = () => (
               </div>
             ))}
           </div>
+          <div className={f('links-column')}>
+            <div className={f('header')}>
+              <Text id="footer.listen-us" />
+            </div>
+            {PODCASTS_PLATFORMS.map(({ id, label, link }) => (
+              <div key={id} className={f('item')}>
+                <ExternalLink key={id} href={link}>
```
Comment (Drapegnik) [inline components/common/layout/Footer.js]:
```suggestion
                <ExternalLink href={link}>
```

File: constants/social.js [hunk 5]
```diff
@@ -75,6 +75,31 @@ export const SHARE_NETWORKS = [
...
+  {
+    id: 'googlepodcasts',
+    label: 'Google Podcasts',
+    link: 'https://podcasts.google.com/?feed=aHR0cHM6Ly93aXIuYnkvcnNzL3BvZGNhc3RzLw',
+  },
+  {
+    id: 'vkpodcasts',
+    label: 'VK',
+    // TODO: upd vk link.
+    link: 'https://vk.com/wir_by',
```
Comment (Drapegnik) [inline constants/social.js]:
```suggestion
    link: 'https://vk.com/podcasts-160609284',
```

File: constants/social.js [hunk 6]
```diff
@@ -75,6 +75,31 @@ export const SHARE_NETWORKS = [
...
+  {
+    id: 'applepodcasts',
+    label: 'Apple Podcasts',
+    link:
+      'https://podcasts.apple.com/by/podcast/wir-by-беларуская-і-сусветная-культура/id1511316106',
+  },
+  {
+    id: 'googlepodcasts',
+    label: 'Google Podcasts',
+    link: 'https://podcasts.google.com/?feed=aHR0cHM6Ly93aXIuYnkvcnNzL3BvZGNhc3RzLw',
```
Comment (Drapegnik) [inline constants/social.js]:
```suggestion
    link: 'https://podcasts.google.com/feed/aHR0cHM6Ly93aXIuYnkvcnNzL3BvZGNhc3RzLw',
```

File: pages/collection.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+import { renderNodeList } from 'utils/formatters';
+import host from 'utils/host';
+
+const b = bem(styles);
+
+const mapStateToProps = (state, { lang }) => collectionsSelectors.getData(state, lang);
+
+const CollectionPage = ({
+  lang,
+  routerQuery: { slug: _ },
```
Comment (Drapegnik) [inline pages/collection.js]: unused?

File: redux/ducks/collections.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
+import createReducer from 'type-to-reducer';
+
```
Comment (Drapegnik) [inline redux/ducks/collections.js]:
```suggestion
```

File: redux/ducks/collections.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  const { collection } = getState(state);
+  const localizedArticles = getLocalizedArticles(collection.articles, lang);
+
+  const authors = Object.values(
+    localizedArticles.reduce((acc, { tagsByTopic: { authors: articleAuthors } }) => {
+      articleAuthors.forEach(authorTag => {
+        acc[authorTag.slug] = authorTag;
+      });
+      return acc;
+    }, {})
```
Comment (Drapegnik) [inline redux/ducks/collections.js:43]: alternative way:
```js
const authors = new Set();
localizedArticles.forEach(({ tagsByTopic }) => {
   tagsByTopic.authors.forEach(tag => {
       authors.add(tag);
   });
});
```

> [TL:DR Set is almost two times faster than Object](https://stackoverflow.com/questions/44243103/javascript-set-vs-array-vs-object-definition)

File: redux/ducks/collections.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  );
+
+  return {
+    collection: getLocalizedCollection(omit(collection, ['articles']), lang),
```
Comment (Drapegnik) [inline redux/ducks/collections.js:47]: why omit?

File: routes.js [hunk 11]
```diff
@@ -50,6 +50,11 @@ const ROUTES = [
     name: 'tag',
     pattern: `topic/:topic(${topics})/tag/:tag`,
   },
+  {
+    name: 'collection',
+    pattern: 'collection/:slug',
+    page: 'collection',
```
Comment (Drapegnik) [inline routes.js]:
```suggestion
```

File: utils/getters.js [hunk 12]
```diff
@@ -23,7 +23,7 @@ export const getLocalizedCollection = (
   articleIndex,
   name: localize(name, lang),
   description: localize(description, lang),
-  articles: articles.map(a => getLocalizedArticle(a, lang)), // eslint-disable-line no-use-before-define
+  articles: articles?.map(a => getLocalizedArticle(a, lang)), // eslint-disable-line no-use-before-define
```
Comment (Drapegnik) [inline utils/getters.js:26]: maybe add optional param `omitArticles`?

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: components/articles/cards/ArticleCard.js (hunk 1 above)
Comment (UladBohdan) [reply components/articles/cards/ArticleCard.js:73]: I guess there is a way to use css only and somehow resolve long titles there, but I don't really want to do that.

File: components/articles/cards/ArticleCard.js (hunk 2 above)
Comment (UladBohdan) [reply components/articles/cards/ArticleCard.js]: Done.

File: components/common/layout/Footer.js (hunk 3 above)
Comment (UladBohdan) [reply components/common/layout/Footer.js:41]: We are replacing this block with "podcasts platforms" block but I guess we might want to quickly switch it back again.

File: components/common/layout/Footer.js (hunk 4 above)
Comment (UladBohdan) [reply components/common/layout/Footer.js]: Done.

File: redux/ducks/collections.js (hunk 8 above)
Comment (UladBohdan) [reply redux/ducks/collections.js]: Done.

File: redux/ducks/collections.js (hunk 9 above)
Comment (UladBohdan) [reply redux/ducks/collections.js:43]: I don't think it is a good idea to rely on uniqueness of objects (it's much more reliable to use slugs as keys). Also it does not seem to work: I've tried and it looks like object uniqueness is something different from deep value equality (probably addresses or smth like that).
Screen Shot 2020-07-04 at 17 38 36

File: redux/ducks/collections.js (hunk 10 above)
Comment (UladBohdan) [reply redux/ducks/collections.js:47]: Collection contains articles, each article contain reference to collection. So we are localizing recursively here, that is not necessary. Articles are delivered separatly as blocks.

File: routes.js (hunk 11 above)
Comment (UladBohdan) [reply routes.js]: Done.

File: constants/social.js (hunk 5 above)
Comment (UladBohdan) [reply constants/social.js]: Done.

File: constants/social.js (hunk 6 above)
Comment (UladBohdan) [reply constants/social.js]: Done.

File: utils/getters.js (hunk 12 above)
Comment (UladBohdan) [reply utils/getters.js:26]: How is it better than just omitting them if they are not present?

File: pages/about.js [hunk 13]
```diff
@@ -61,12 +62,11 @@ const AboutPage = ({ lang }) => (
           />
         ))}
       </div>
-      <div className={b('text', { 'margin-top': 'medium' })}>
-        <Thanks1 />
-      </div>
-      <div className={b('text', { 'margin-top': 'medium' })}>
-        <Thanks2 />
-      </div>
+      {times(3).map(idx => (
```
Comment (Drapegnik) [inline pages/about.js]:
```suggestion
      {Array.from({ length: 3 }).map((_, idx) => (
```

**Review by Drapegnik — CHANGES_REQUESTED** [review]

**Review by Drapegnik — DISMISSED** [review]

**Pull request merged**